check.py

```diff
@@ -93,43 +121,44 @@ def setup_waterfall():
   except Exception, e:
     warn('could not run vanilla LLVM from waterfall: ' + str(e) + ', looked for clang at ' + CLANG)
 
-fetch_waterfall()
-setup_waterfall()
+if options.test_waterfall:
+  fetch_waterfall()
+  setup_waterfall()
 
-if only_prepare:
+if options.only_prepare:
   print 'waterfall is fetched and setup, exiting since --only-prepare'
   sys.exit(0)
 
 # external tools
 
 has_node = False
 try:
-  subprocess.check_call(['nodejs', '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  subprocess.check_call(['nodejs', '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
```

Comment:
The issue is with the suffixes, which may be `emcc`, `emcc.bat`, or `nodejs.exe`, which this doesn't specify. The suffixes are autodetected only if `shell=True`, otherwise it does not look for `nodejs.exe` when asking to run `nodejs`.